rust/foxglove/src/websocket/service.rs

```diff
@@ -31,6 +31,14 @@ impl ServiceId {
     pub fn new(id: u32) -> Self {
         Self(id)
     }
+
+    /// Allocates the next service ID.
+    pub fn next() -> Self {
+        static NEXT_ID: AtomicU32 = AtomicU32::new(1);
+        let id = NEXT_ID.fetch_add(1, Ordering::Relaxed);
+        assert_ne!(id, 0);
```

Comment:
maybe add a bit more detail:
```suggestion
        assert_ne!(id, 0, "ServiceId overflowed");
```